src/python/pants/scm/git.py

```diff
@@ -567,27 +569,28 @@ def _read_tree(self, path):
       return tree
     tree = {}
     object_type, tree_data = self._read_object_from_repo(rev=self.rev, relpath=path)
-    assert object_type == 'tree'
+    assert object_type == b'tree'
     # The tree data here is (mode ' ' filename \0 20-byte-sha)*
     i = 0
     while i < len(tree_data):
       start = i
-      while tree_data[i] != ' ':
+      while tree_data[i] != b' ':
         i += 1
-      mode = tree_data[start:i]
+      mode = b''.join(tree_data[start:i])
       i += 1  # skip space
       start = i
       while tree_data[i] != NUL:
         i += 1
-      name = tree_data[start:i]
-      sha = tree_data[i + 1:i + 1 + GIT_HASH_LENGTH].encode('hex')
+      name = b''.join(tree_data[start:i])
+      sha = b''.join(tree_data[i + 1:i + 1 + GIT_HASH_LENGTH])
+      sha_hex = binascii.hexlify(sha)
```

Comment:
Original had a couple problems:
- calling `encode()` on a list
- using `'hex'` encoding, which has been dropped from the `encode()` function in Py3

Instead, `binascii.hexlify()` will get the result we want, and it must be passed a byte string instead of a list.